### ISSUE TITLE
Roslynator seems to run on code generated by source generator and does not read editorconfig settings while doing so

### ISSUE THREAD

**Issue by DavidZidar** [JosefPihrt/Roslynator#876]
I have a project that references another project that has source generators, this seems to cause a bug in Roslynator v4.0.0 that makes it print incorrect warnings about missing configuration options even though they are present in the editorconfig file.

I have created a sample project that reproduces the issue:
https://github.com/DavidZidar/roslynator-analyzer-bug

Example log output:
```
> dotnet build

Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  TestAnalyzers -> C:\code\test\roslynator-bug\src\TestAnalyzers\bin\Debug\net6.0\TestAnalyzers.dll
CSC : warning ROS0003: Analyzer RCS1018 requires config option to be specified: roslynator_accessibility_modifiers [C:\code\test\roslynator-bug\src\TestProject\TestProject.csproj]
CSC : warning ROS0003: Analyzer RCS1096 requires config option to be specified: roslynator_enum_has_flag_style [C:\code\test\roslynator-bug\src\TestProject\TestProject.csproj]
  TestProject -> C:\code\test\roslynator-bug\src\TestProject\bin\Debug\net6.0\TestProject.dll

Build succeeded.

CSC : warning ROS0003: Analyzer RCS1018 requires config option to be specified: roslynator_accessibility_modifiers [C:\code\test\roslynator-bug\src\TestProject\TestProject.csproj]
CSC : warning ROS0003: Analyzer RCS1096 requires config option to be specified: roslynator_enum_has_flag_style [C:\code\test\roslynator-bug\src\TestProject\TestProject.csproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.91
```

**Product and Version Used**: Roslynator.Analyzers v4.0.0

**Steps to Reproduce**:
1. Configure required Roslynator settings in editorconfig-file.
2. Reference a project containing source generators that are generating any code.
3. Build.

**Actual Behavior**:
Incorrect warnings are printed.
```csharp
CSC : warning ROS0003: Analyzer RCS1018 requires config option to be specified: roslynator_accessibility_modifiers [C:\code\test\roslynator-bug\src\TestProject\TestProject.csproj]
```

**Expected Behavior**:
No warnings should be printed because the configuration options are in fact present.

**Comment by JosefPihrt** [comment]
Hi,

thanks for the report and code sample.

It seems that code analysis is executed on files generated with source generator. This behavior is not specific to Roslynator analyzers. If you use this code:

```cs
class C
{
}
```

in your sample project you will get following error:

```
warning CA1812: C is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make it static (Shared in Visual Basic).
```

The other thing I discovered is that the file path of the generated file is relative which make is incompatible with editorconfig (which requires absolute path to determine proper configuration)

This can be fixed by verifying if the source file path is relative or absolute and skip these files.

Current solution to this issue is to mark generated source as generated. Either set `hintName` to `TestSource.Generated` or similar or add  auto-generated marker at the beginning of the source:

```
// <auto-generated>
```

Then the generated source will be excluded from code analysis.

Another option is to disable `ROS0003`.

I would be interested if Roslyn team intended to execute code analysis on source generator's files and if there is some way how to configure this behavior.

**Comment by DavidZidar** [comment]
Oh, interesting, thank you that information. I didn't know that `.Generated` in the file name or that comment did anything special. That takes care of source code I generate myself, but unfortunately the source generators in `Microsoft.Toolkit.Mvvm` which I also use doesn't do either of those things so I'm still getting the warning even if I fix my own code. I can certainly report an issue to that team and hope that they'll fix this on their end.

Regarding the relative path, would it be possible to resolve the path relative to the project and check if the file exists on disk? I'm thinking maybe it wouldn't make sense to look for an editorconfig file for a virtual file.

Whether the Roslyn team intended to execute code analysis on source generator's files or not I can only guess, but I can imagine it being useful occasionally, like checking for nullability issues and such rather than stylistic issues.

**Comment by PhilPJL** [comment]
Hmmm, how do I disable ROS0003? 

This doesn't work
![image](https://user-images.githubusercontent.com/271656/150696564-156640fc-f17a-47a9-b0d1-bc3655661ed7.png)

![image](https://user-images.githubusercontent.com/271656/150696574-a53597eb-27b4-438e-b11e-bf6190233fbf.png)

**Comment by JosefPihrt** [comment]
Following entry
```editorconfig
dotnet_diagnostic.ROS0003.severity = none
```

will disable it but there are some false positives. Until it is resolved my recommendation is to suppress it:
```editorconfig
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Roslynator", "ROS0003")]
```


I will probably disable by default analyzers that require option to be set. Which means that by default you should not see any warnings.

Another option is to disable ROS0003 by default.

**Comment by PhilPJL** [comment]
ok thanks, that's sorted it
